### PR TITLE
a way to merge multiple in-memory embedding stores

### PR DESCRIPTION
**Pull request by ghost**
## Summary by CodeRabbit

- **New Features**
	- Introduced enhanced flexibility for managing entries within the `InMemoryEmbeddingStore`.
- **Tests**
	- Added a test to verify the functionality of merging multiple `InMemoryEmbeddingStore` instances.

### CODE REVIEW

**Comment by geoand** [comment]
Where is this useful?

**Comment by langchain4j** [comment]
Had a discussion with someone, they wanted to keep different files for different data, but be able to merge them all into one embedding store in memory. Not sure this is widely useful

**Comment by geoand** [comment]
I see no harm in it, but like you said, it's probably not terribly useful

File: langchain4j/src/main/java/dev/langchain4j/store/embedding/inmemory/InMemoryEmbeddingStore.java [hunk 1]
```diff
@@ -163,6 +172,27 @@ public static InMemoryEmbeddingStore<TextSegment> fromFile(String filePath) {
         return fromFile(Paths.get(filePath));
     }
 
+    /**
+     * Merges given {@code InMemoryEmbeddingStore}s into a single {@code InMemoryEmbeddingStore},
+     * copying all entries from each store.
+     */
+    public static <Embedded> InMemoryEmbeddingStore<Embedded> merge(Collection<InMemoryEmbeddingStore<Embedded>> stores) {
+        List<Entry<Embedded>> entries = new ArrayList<>();
```
Comment (Wadimz) [inline langchain4j/src/main/java/dev/langchain4j/store/embedding/inmemory/InMemoryEmbeddingStore.java:181]:
```suggestion
        ensureNotNull(stores, "stores");
        List<Entry<Embedded>> entries = new ArrayList<>();
```

**Review by Wadimz — APPROVED** [review]

**Review by Wadimz — APPROVED** [review]

**Pull request merged**